tests/e2e/privacy.spec.js

```diff
@@ -56,50 +63,143 @@ describe('Privacy', () => {
+
+    it('does not block tracker requests on the page', async () => {
+      await disableCache();
+      await browser.url(PAGE_URL);
+
+      await switchToPanel(async () => {
+        await getExtensionElement('button:detailed-view').click();
+
+        for (const trackerId of TRACKER_IDS) {
+          const trackerEl = await getExtensionElement(
+            `button:tracker:${trackerId}`,
+          );
+
+          if (trackerEl.isDisplayed()) {
+            await expect(
+              getExtensionElement(`icon:tracker:${trackerId}:blocked`),
+            ).not.toBeDisplayed();
+            await expect(
+              getExtensionElement(`icon:tracker:${trackerId}:modified`),
+            ).not.toBeDisplayed();
+          }
+        }
+      });
+    });
+
+    it('blocks tracker requests on the page', async () => {
+      await updatePrivacySettings('anti-tracking', true);
+
+      await disableCache();
+      await browser.url(PAGE_URL);
+
+      await switchToPanel(async () => {
+        await getExtensionElement('button:detailed-view').click();
+
+        for (const trackerId of TRACKER_IDS) {
+          const trackerEl = await getExtensionElement(
+            `button:tracker:${trackerId}`,
+          );
+
+          if (trackerEl.isDisplayed()) {
+            await expect(
+              getExtensionElement(`icon:tracker:${trackerId}:blocked`),
+            ).toBeDisplayed();
+          }
+        }
+      });
+    });
+  });
+
+  describe('Regional Filters', () => {
+    beforeEach(() => updatePrivacySettings('regional-filters', false));
+
+    const WEBSITE_URL = 'https://www.cowwilanowie.pl/';
+    const SELECTOR = '.a-slider';
+
+    it('shows the ads on the page', async () => {
       await browser.url(WEBSITE_URL);
+      await expect($(SELECTOR)).toBeDisplayed();
+    });
+
+    it('hides the ads on the page', async () => {
+      await updatePrivacySettings('regional-filters', true);
+      const checkbox = await getExtensionElement(
+        'checkbox:regional-filters:pl',
+      );
 
-      for (const ad of await $$(SELECTOR)) {
-        await expect(ad).not.toBeDisplayed();
+      if (!(await checkbox.getProperty('checked'))) {
+        await checkbox.click();
+        await expect(checkbox).toBeChecked();
       }
+
+      await browser.url(WEBSITE_URL);
+      await expect($(SELECTOR)).not.toBeDisplayed();
     });
   });
 
-  describe('Global pause', () => {
-    const WEBSITE_URL = 'https://www.onet.pl/';
+  describe('Pause Website', () => {
+    it("pauses the website's privacy settings", async () => {
+      await browser.url(PAGE_URL);
+
+      await switchToPanel(async () => {
+        const pauseComponent = await getExtensionElement('component:pause');
+        const pauseButton = await getExtensionElement('button:pause');
+
+        if (await pauseComponent.getProperty('paused')) {
+          await pauseButton.click();
+          await browser.pause(1000);
```

Comment:
can we await some ui change here, like a class name added?

Reply:
I checked, and we wait for the result on the panel itself, so the pause here is not needed.